Lib/textwrap.py

```diff
@@ -199,6 +199,23 @@ def _fix_sentence_endings(self, chunks):
             else:
                 i += 1
 
+    def _find_width_index(self, text, width):
+        """_find_width_index(text : string, width: int)
+
+        Find at which index the text has the required width.
+        """
+        # In most cases text_len will just use the number of characters, so this heuristic prevents calculating width
+        # for each character
+        if self.text_len(text[:width]) == width:
+            # For character widths greater than one, width can be more than the number of characters
+            return min(width, len(text))
+        cur_text = ''
+        for i, c in enumerate(text):
+            cur_text += c
+            cur_width = self.text_len(cur_text)
+            if cur_width >= width:
+                return i+1
```

Comment:
What should happen if the next character has width 4? Example:

```python
_find_width_index('1234Q5678', 4)
```

Should this be 4 (`['1234', 'Q5678']`) or 5  (`['1234Q', '5678']`)?

I slightly lean towards the greedy mechanism:

```python
if cur_width > width:
    return i
```

Reply:
You're right, that makes more sense, I'll add a test to indicate a case where using the previous solution would actually be incorrect. I'll also add something to ensure always at least one character is added and add a test for that also.